test/functional/tests/fault_injection/test_fault_injection_core_in_raid.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet
+from storage_devices.raid import Raid, RaidConfiguration, MetadataVariant, Level
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeSet([DiskType.sata, DiskType.hdd]))
+def test_fault_injection_core_in_raid(cache_mode):
+    """
+        title: Test if OpenCAS reject using CAS HDD member to build SW RAID.
+        description: |
+          Test if OpenCAS handle properly attempting of use CAS HDD member to build SW RAID.
+        pass_criteria:
+          - Expected to reject operation with proper warning.
+    """
+    with TestRun.step("Prepare CAS device."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_disk.create_partitions([Size(2, Unit.GibiByte), Size(2, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_dev = core_disk.partitions[0]
+        second_core_dev = core_disk.partitions[1]
+
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Attempt to use CAS HDD member to build SW RAID."):
+        raid_disk_1 = core_dev
+        raid_disk_1_partition_id = core_dev.system_path.split('/')[-1]
+        raid_disk_2 = second_core_dev
+
+        expected_msg_1 = f"{raid_disk_1_partition_id} is not suitable for this array".lower()
+        expected_msg_2 = f"Cannot open {core_dev.system_path}: Device or resource busy".lower()
+
+        config = RaidConfiguration(
+            level=Level.Raid1,
+            metadata=MetadataVariant.Legacy,
+            number_of_devices=2,
+            size=Size(20, Unit.GiB))
+
+        try:
+            raid = Raid.create(config, [raid_disk_1, raid_disk_2])
+            TestRun.LOGGER.error(f"RAID created successfully. Expected otherwise.")
+        except Exception as ex:
+            exception_output = str(ex).lower()
+            if (expected_msg_1 in exception_output) or (expected_msg_2 in exception_output):
+                TestRun.LOGGER.info("RAID not created. "
+                                    "Found expected warning in exception message.")
+            else:
+                TestRun.LOGGER.error(f"RAID not created but warning message not as expected.\n"
+                                     f"Actual: '{exception_output}'.\n"
+                                     f"Expected: '{expected_msg_1}' or '{expected_msg_2}'.")
+    with TestRun.step("Stop cache."):
+        cache.stop()
```

Comment:
AFAIR we don't need to stop cache manually after test. Teardown should do the job. Am I right, @Deixx?